sympy/concrete/summations.py

```diff
@@ -1194,3 +1285,53 @@ def eval_sum_hyper(f, i_a_b):
                     return S.NegativeInfinity
             return None
         return Piecewise(res, (old_sum, True))
+
+
+def _eval_matrix_sum(expression):
+    f = expression.function
+    for n, limit in enumerate(expression.limits):
+        i, a, b = limit
+        dif = b - a
+        if dif.is_Integer:
+            if (dif < 0) == True:
+                a, b = b + 1, a - 1
+                f = -f
+
+            newf = eval_sum_direct(f, (i, a, b))
+            if newf is not None:
+                return newf.doit()
+
+
+def dummy_with_inherited_properties(limits):
```

Comment:
This looks useful. How about
```python
def properties(*values):
    valid = {}
    for a in assumptions or _assume_defined:
        if all(getattr(v, 'is_' + a, None) for v in values):
            valid[a] = True
   return valid
```
As for this routine...like `solveset` the assumptions of the passed symbols should be irrelevant as they are like dummy symbols. So if I do `x = Dummy(negative=True); Integral(x, (x, 1, 3))` I should get the same result as for `Integral(y, (y, 1, 3))`...unless we want to start raising errors if the symbols have assumptions that conflict with the limits. Assuming assumptions are irrelevent (except for this assumption that they are so), you should not keep assumptions which are already True since they may conflict with the assumptions that aren't provided but *are* true based on the limits.

What about:
```python
def consistent_dummy(x, *values):
    valid = properties(*values)
    xa = x._assumptions
    violations = any(k in valid for k for k in x._assumptions if xa[k] is False)
    complete = all(xa[k] for k in valid)
    if violations or not complete:
        return Dummy(x.name, **valid)
```
Neither of these was tested...proceed with caution.

Reply:
> unless we want to start raising errors if the symbols have assumptions that conflict with the limits

I do not see this as a drawback. Except that it actually doesn't happen right now. At least it works well to define variables as integer and still use oo as a limit (which is why I inherited from the provided symbol in the first place, as oo is not integer, there would be no integer assumption on the Dummy, not that I know if it matters, but it makes sense to have integer assumptions on summation variables...)

I'm not sure how your code is supposed to work. How do we provide which assumptions to check for? Provide an additional argument (or two? `_assume_defined`) to properties and consistent_dummy?

I'm sure that there are better ways to write this. I've been busy getting it to work and battling other peculiarities in the current code base. Right now, a new simplify-step that effects Piecewise being merged earlier today...